uaclient/tests/test_security.py

```diff
@@ -941,13 +941,19 @@ def test_raise_userfacing_error_on_invalid_usn_metadata(
         FakeConfig,
     ):
         with pytest.raises(exceptions.SecurityAPIMetadataError) as exc:
-            prompt_for_affected_packages(
-                cfg=FakeConfig(),
-                issue_id="USN-###",
-                affected_pkg_status=affected_pkg_status,
-                installed_packages=installed_packages,
-                usn_released_pkgs=usn_released_pkgs,
-            )
+            with mock.patch("uaclient.util.sys") as m_sys:
+                m_stdout = mock.MagicMock()
+                type(m_sys).stdout = m_stdout
+                type(m_stdout).encoding = mock.PropertyMock(
+                    return_value="utf-8"
```

Comment:
Now I finally discern what mock @renanrodrigo was talking about. Didn't realize you were talking about this PR earlier. +1